src/buildSrc/src/main/java/org/niis/xroad/oasvalidatorplugin/Oas3ValidatorGradlePlugin.java

```
@@ -0,0 +1,78 @@
+ * THE SOFTWARE.
+ */
+package org.niis.xroad.oasvalidatorplugin;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.openapi4j.core.exception.ResolutionException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Oas3ValidatorGradlePlugin implements Plugin<Project> {
+    private static final String PARAMETERS_NAME = "apiValidationParameters";
+    private static final String TASK_NAME = "validateApiDefinitions";
+
+    public Oas3ValidatorGradlePlugin() {
+    }
+
+    @Override
+    public void apply(Project project) {
+        Oas3ValidatorExtension extension = project.getExtensions()
+                .create(PARAMETERS_NAME, Oas3ValidatorExtension.class);
+
+        project.task(TASK_NAME)
+                .doLast(task -> {
+                    if (extension.getApiDefinitionPaths() == null || extension.getApiDefinitionPaths().isEmpty()) {
+                        throw new GradleException("No API definition file paths provided");
+                    }
+                    System.out.println(System.lineSeparator() + "--- API VALIDATION START ---");
+                    boolean isCompleteValidationSuccess = true;
+                    List<ApiValidationResults> allValidationResults = new ArrayList<>();
+                    extension.getApiDefinitionPaths().forEach(path -> {
+                        try {
+                            ApiValidationResults validationResults = Oas3Validator.validate(path);
+                            allValidationResults.add(validationResults);
+                        } catch (ResolutionException e) {
+                            throw new GradleException("API definition not found", e);
+                        }
+                    });
+                    for (ApiValidationResults apiValidationResults : allValidationResults) {
+                        ApiValidationResult apiSpecValidationResult = apiValidationResults
```

There seem to be some validation error cases (duplicate property or required -tag in openapi-authentication.yaml) that result in an exception that is printed out, but validation result is SUCCESS and build continues.

Possibly caused by a JsonParseException that is thrown out instead of GradleException. Not sure, but would it be a good idea to catch all exceptions that happen during validation and wrapping them in GradleException("API definition validation failed", rootCause), or something?

---

Pushed a fix for this. All Exceptions are now caught in the Plugin. The JsonParseException was not actually thrown but only logged by one of the parser libraries. But the data was still recoverable and is now added to the validation failure data.